recipes/instinct-cpp/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class InstinctCppConan(ConanFile):
+    name = "instinct-cpp"
+    description = ("instinct.cpp provides ready to use alternatives to OpenAI Assistant API"
+                   "and built-in utilities for developing AI Agent applications (RAG, Chatbot, Code interpreter) powered by language models. "
+                   "Call it langchain.cpp if you like.")
+    license = "Apache-2.0"
+    url = "https://github.com/RobinQu/instinct.cpp"
+    homepage = "https://github.com/RobinQu/instinct.cpp"
+    topics = ("llm", "agent", "genai")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_duckdb": [True, False],
+        "with_exprtk": [True, False],
+        "with_pdfium": [True, False],
+        "with_duckx": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_duckdb": True,
+        "with_exprtk": True,
+        "with_pdfium": True,
+        "with_duckx": True,
+        "duckdb/*:with_httpfs": True,
+        "icu/*:with_extras": True,
+        "icu/*:data_packaging": "static"
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "15",
+            "clang": "17",
+            "gcc": "12"
```

Comment:
Has this library been tested on Windows? we would add a min msvc if it's possible to run there :)